File: 581-shortest-unsorted-continuous-subarray/581-shortest-unsorted-continuous-subarray.cs
public class Solution {
    public int FindUnsortedSubarray(int[] nums) {
        int l = nums.Length;
        var a = new int[l];
        nums.CopyTo(a,0);
        Array.Sort(a);
        int i=0,j=l-1;
        while(i<l && nums[i]==a[i]){
            i++;
        }
        while(j>i && nums[j]==a[j]){
            j--;
        }
        return j-i+1;
    }
}

File: 581-shortest-unsorted-continuous-subarray/581-shortest-unsorted-continuous-subarray.cpp
class Solution {
public:
    int findUnsortedSubarray(vector<int>& nums) {
        vector<int> a(nums);
        sort(a.begin(),a.end());
        int n = nums.size(), i=0,j=n-1;
        while (i<n && nums[i]==a[i])
        {
            i++;
        }
        while (j>i && nums[j]==a[j])
        {
            j--;
        }
        return j-i+1;
    }
};